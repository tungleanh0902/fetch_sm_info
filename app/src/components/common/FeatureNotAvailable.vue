<template>
  <TmDataMsg icon="timeline">
    <div slot="title">Feature not available</div>
    <div slot="subtitle">
      We haven't implemented this feature for this network yet.
      <br />
      <br />
      <a class="intercom-button" @click="$store.dispatch(`displayMessenger`)"
        >Reach out if you'd like us to add support for this feature.</a
      >
    </div>
  </TmDataMsg>
</template>

<script>
import TmDataMsg from "common/TmDataMsg"
export default {
  name: `feature-not-available`,
  components: { TmDataMsg },
  props: {
    feature: {
      type: String,
      required: true,
    },
  },
}
</script>
