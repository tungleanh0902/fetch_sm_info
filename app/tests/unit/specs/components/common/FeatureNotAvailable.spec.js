import { mount } from "@vue/test-utils"
import FeatureNotAvailable from "common/FeatureNotAvailable"

describe(`FeatureNotAvailable`, () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(FeatureNotAvailable, {
      propsData: {
        feature: "Spacetravel",
      },
    })
  })

  it(`shows a warning about a feature not being available`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
