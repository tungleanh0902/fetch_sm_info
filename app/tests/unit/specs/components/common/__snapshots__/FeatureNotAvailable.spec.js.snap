// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureNotAvailable shows a warning about a feature not being available 1`] = `
<div
  class="tm-data-msg"
>
  <div
    class="tm-data-msg__icon"
  >
    <i
      class="material-icons notranslate"
    >
      timeline
    </i>
  </div>
   
  <div
    class="tm-data-msg__text"
  >
    <h2
      class="tm-data-msg__title"
    >
      <div>
        Feature not available
      </div>
    </h2>
     
    <h3
      class="tm-data-msg__subtitle"
    >
      <div>
        
    We haven't implemented this feature for this network yet.
    
        <br />
         
        <br />
         
        <a
          class="intercom-button"
        >
          Reach out if you'd like us to add support for this feature.
        </a>
      </div>
    </h3>
  </div>
</div>
`;
